src/query/test/pseudo.ts

```diff
@@ -516,6 +516,59 @@ t.test('specificity calculation for pseudo selectors', async t => {
   )
 })
 
+t.test(':path selector', async t => {
+  const simpleGraph = getSimpleGraph()
+  const all = {
+    edges: new Set(simpleGraph.edges),
+    nodes: new Set(simpleGraph.nodes.values()),
+  }
+  const empty: GraphSelectionState = {
+    edges: new Set(),
+    nodes: new Set(),
+  }
+
+  const queryToExpected = new Set<TestCase>([
+    // Test basic path matching - all patterns must be quoted
+    [
+      ':path("*")',
+      all,
+      ['my-project', 'a', 'b', 'c', 'd', 'e', 'f', '@x/y'],
+    ], // Match all
+    [
+      ':path("node_modules/**")',
+      all,
+      ['a', 'b', 'c', 'd', 'e', 'f', '@x/y'],
+    ], // Match deps in node_modules
+    [
+      ':path("**/.vlt/**")',
+      all,
+      ['a', 'b', 'c', 'd', 'e', 'f', '@x/y'],
+    ], // Match vlt store paths
+    [':path("**/my-project**")', all, ['my-project']], // Match root project
+    [':path("**/a/**")', all, ['a']], // Match specific package
+    [':path("**/b/**")', all, ['b']], // Match specific package 'b'
+    [':path("**/doesnotexist/**")', all, []], // No matches
+
+    // Test edge cases
+    [':path("")', all, []], // Empty pattern should match nothing
+    [':path()', all, []], // Missing pattern should match nothing
+
+    // Test from empty partial
+    [':path("*")', empty, []], // Empty input should return empty
+  ])
+
+  for (const [query, partial, expectedNames] of queryToExpected) {
+    try {
+      await testPseudo(query, all, partial, false, expectedNames)
+    } catch (error) {
+      t.fail(`Failed query "${query}": ${error}`)
+    }
+  }
+})
```

Comment:
Let's rewrite this test to use the new `getPathBasedGraph` fixture once that has been added to the `src/query/test/fixtures/graph.ts` file.